### PR TITLE
Switch to self-hosted pool to fix Android CI

### PR DESCRIPTION
**Pull request by skyline75489**
* Switch to internal pool to avoid being hit by GHA image updates.
* Add script to install Android emulator if needed.

### CODE REVIEW

File: .github/workflows/android-build.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     #       it doesn't work on macos-14.
     #         HVF error: HV_UNSUPPORTED
     #       it works on macos-13 but with macos-15 being released soon that isn't a long term solution.
-    runs-on: ubuntu-latest
+    runs-on: [ "self-hosted", "1ES.Pool=onnxruntime-genai-Ubuntu2204-AMD-CPU" ]
```
Comment (skyline75489) [inline .github/workflows/android-build.yml:24]: This pool is also used in Linux x64 CI

File: .github/workflows/android-build.yml [hunk 2]
```diff
@@ -40,48 +40,38 @@ jobs:
...
           echo "ANDROID_NDK_HOME=$ANDROID_NDK_HOME"
           echo "ANDROID_NDK_LATEST_HOME=$ANDROID_NDK_LATEST_HOME"
           ls -l $ANDROID_HOME/ndk
 
+      - name: Check if emulator is installed and install it if needed
+        run: |
+          if [ -d "${ANDROID_SDK_ROOT}/emulator" ]; then
+              echo "${ANDROID_SDK_ROOT}/emulator exists"
+          else
+              ${ANDROID_SDK_ROOT}/cmdline-tools/latest/bin/sdkmanager --install "emulator"
```
Comment (skyline75489) [inline .github/workflows/android-build.yml:62]: This is from https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/azure-pipelines/templates/use-android-emulator.yml

File: .github/workflows/android-build.yml [hunk 3]
```diff
@@ -40,48 +40,38 @@ jobs:
...
+          fi
+
       # Needed for linux
       - name: Install jq
         run: |
           sudo apt-get install jq
 
-      - uses: actions/setup-dotnet@v4
```
Comment (skyline75489) [inline .github/workflows/android-build.yml:57]: For some unknown reason, `dotnet install` failed with permission denied.

See https://github.com/microsoft/onnxruntime-genai/actions/runs/12385746487/job/34573153027

File: .github/workflows/android-build.yml [hunk 4]
```diff
@@ -40,48 +40,38 @@ jobs:
...
-        with:
-          dotnet-version: '8.0.x'
-
       - name: Checkout OnnxRuntime GenAI repo
         uses: actions/checkout@v4
         with:
           submodules: true
 
-      - name: Get the Latest OnnxRuntime Nightly Version
```
Comment (skyline75489) [inline .github/workflows/android-build.yml:66]: Removing these steps has no effect on either building or testing GenAI on Android platform.

For building, without explicitly pass `--ort_home`, the version in [ortlib.cmake](https://github.com/microsoft/onnxruntime-genai/blob/main/cmake/ortlib.cmake) will be used.

For testing, the latest release version of ORT is used:

https://github.com/microsoft/onnxruntime-genai/blob/c77e525243dc233b88edaefabbfc5f19facf4044/src/java/src/test/android/app/build.gradle#L44

I remember @baijumeswani  mentions that the nightly version of ORT should be used for testing, but I'm not suer that's available for Android.

**Review by baijumeswani — APPROVED** [review]

**Comment by baijumeswani** [comment]
Thanks 👍🏽

**Pull request merged**